src/plugins/intel_cpu/src/nodes/kernels/stack_allocator.hpp

```diff
@@ -0,0 +1,313 @@
+            return alloc;
+        }
+    }
+
+    void deallocate() {
+        while (!allocations.empty()) {
+            const auto& last = allocations.back();
+            if (last->is_used) {
+                break;
+            }
+            current_offset -= last->size;
+            allocations.pop_back();
+        }
+    }
+
+    void checkUnique(bool isCtor) {
+        static thread_local bool isCreated = false;
+        if (isCtor) {
+            if (isCreated) {
+                IE_THROW() << "There should be only one instance of StackAllocator per thread !!";
+            }
+            isCreated = true;
+        } else {
+            isCreated = false;
+        }
+    }
+
+    x64::jit_generator& code_generator;
+    const Xbyak::Reg& base_pointer;
+
+    size_t offset{};
+    size_t current_offset{};
+    std::deque<Allocation::Ptr> allocations{};
+};
+
+class StackAllocator::Address {
+public:
+    Address() = default;
+
+    Address(StackAllocator::Ptr stack_allocator,
+            const size_t alloc_size)
+        : stack_allocator_{stack_allocator}
+        , allocation_{stack_allocator_->allocate(alloc_size)} {
+    }
+
+    virtual ~Address() {
+        release();
+    }
+
+    Address(Address&& addr) noexcept {
+        this->operator=(std::move(addr));
+    }
+
+    Address& operator=(Address&& addr) noexcept {
+        release();
+        stack_allocator_ = std::move(addr.stack_allocator_);
+        allocation_ = std::move(addr.allocation_);
+        return *this;
+    }
+
+    void release() {
+        if (allocation_) {
+            allocation_->is_used = false;
+        }
+        if (stack_allocator_) {
+            stack_allocator_->deallocate();
+        }
+        allocation_ = {};
+        stack_allocator_ = {};
+    }
+
+    operator Xbyak::Address&() {
+        ensureValid();
+        stack_allocator_->commit();
+        return allocation_->address;
+    }
+
+    operator const Xbyak::Address&() const {
+        ensureValid();
+        stack_allocator_->commit();
+        return allocation_->address;
+    }
+
+    virtual Address& operator=(const Xbyak::Xmm& vmm) {
+        stack_mov(*this, vmm);
+        return *this;
+    }
+
+    virtual Address& operator=(const Xbyak::Reg& reg) {
+        stack_mov(*this, reg);
+        return *this;
+    }
+
+    friend void ::ov::intel_cpu::stack_mov(Address& addr, const Xbyak::Xmm& vmm);
+    friend void ::ov::intel_cpu::stack_mov(Address& addr, const Xbyak::Reg& reg);
+    friend void ::ov::intel_cpu::stack_mov(const Xbyak::Xmm& vmm, const Address& addr);
+    friend void ::ov::intel_cpu::stack_mov(const Xbyak::Reg& reg, const Address& addr);
+
+private:
+    void ensureSize(const Xbyak::Reg& reg) const {
+        ensureValid();
+        const size_t reg_size = reg.getBit() / 8;
+        if (reg_size > allocation_->size) {
+            IE_THROW() << "reg size is bigger than space allocated in StackAllocator !!";
+        }
+    }
+
+    void ensureValid() const {
```

Comment:
I think it should be `bool isInitialized()` public method also.

Reply:
Added `isInitialized` and used it inside of `ensureValid`